### ISSUE TITLE
Missing values XGBoost

### ISSUE THREAD

**Issue by NancyChiz** [dmlc/xgboost#10422]
I read that in the latest versions of XGBoost, the model can handle missing values. I am using the model on some data that contains for example, the BMI, bloodpressure, age, binary data (yes/no) and some categorical data, with the target as the number of days. I removed missing values from the target, but kept the rest of the data whether it has missing values or not, as it is necessary(not all people had their blood pressure or BMI measured, but their data is still relevant). When I run the model, I keep getting the error : 
Error in na.fail.default(list(Days = c(5.73055, 15.29236 , : missing values in object. 
But shouldn't this be no problem as it should be able to handle missing values? 

This is some of my code: 

library(caret)
library(xgboost)
library(dplyr)
library(ggplot2)

model <- model[!is.na(model$days), ]

model <- model[!is.na(model$days), ]
trainIndex <- createDataPartition(model[["days"]], p = 0.7, list = FALSE, times = 1)
train_data <- main_attempt[trainIndex, ]
test_data <- main_attempt[-trainIndex, ]

# Define the control method for training
train_control <- trainControl(
  method = "cv",           # k-fold cross-validation
  number = 10,             # number of folds
  verboseIter = TRUE       # display training progress
)

# Define the hyperparameter grid
tune_grid <- expand.grid(
  nrounds = c(50, 100, 150),  # number of boosting rounds
  eta = c(0.01, 0.1, 0.3),    # learning rate
  max_depth = c(3, 6, 9),     # maximum depth of a tree
  gamma = c(0, 0.1, 0.5),     # minimum loss reduction required to make a further partition
  colsample_bytree = c(0.6, 0.8, 1), # subsample ratio of columns when constructing each tree
  min_child_weight = c(1, 5, 10),   # minimum sum of instance weight (hessian) needed in a child
  subsample = c(0.6, 0.8, 1)  # subsample ratio of the training instances
)

# Train the model using caret's train function
set.seed(123)
xgb_model <- train(
  Stay_days ~ .,             # formula for the model
  data = train_data,         # dataset
  method = "xgbTree",        # xgboost algorithm
  trControl = train_control, # control parameters
  tuneGrid = tune_grid,      # hyperparameter grid
  metric = "MAE"             # evaluation metric (Mean Absolute Error)
)
#I get the error here

**Comment by trivialfis** [comment]
I'm a bit confused by the snippets here, not sure what I'm looking at. Also:

>  Error in na.fail.default(list(Days = c(5.73055, 15.29236 , : missing values in object.

I don't know which function is emitting a stop without a backtrace

**Comment by trivialfis** [comment]
Feel free to reopen if you have any further information.
